tests/unit/preprocessor/_other/test_other.py

```diff
@@ -43,5 +44,26 @@ def test_clip(self):
             clip(cube, 10, 8)
 
 
+def test_group_products_string_list():
+    products = [
+        PreprocessorFile(
+            attributes={
+                'project': 'A',
+                'dataset': 'B',
+                'filename': 'A_B.nc'},
+            settings={}),
+        PreprocessorFile(
+            attributes={
+                'project': 'A',
+                'dataset': 'C',
+                'filename': 'A_C.nc'},
```

Comment:
they are essentially the same - do you not want to vary a third parameter e.g. `A_B_x_1990-1991.nc` and `A_C_y_1950-1991.nc`?

Reply:
This test is just to check that _group_products can work with both strings or lists, as it was something thas was missing in the codecov reports:

```
    grouped_by_string = _group_products(products, 'project')
    grouped_by_list = _group_products(products, ['project'])
```
The value of the filenames is not that relevant in this test, it's something that get's checked in other tests in `test_recipe.py`

Reply:
excellent then, cheers for the explanation :+1: